utils/lib/rr_graph_capnp/graph2.py

```diff
@@ -0,0 +1,548 @@
+import os.path
+import re
+from lib.rr_graph import graph2
+from lib.rr_graph import tracks
+
+import capnp
+capnp.remove_import_hook()
+
+CAMEL_CASE_CAPITALS = re.compile('([A-Z]+)')
+
+ENUM_CACHE = {}
+
+
+def enum_from_string(enum_type, s):
+    if s == 'uxsdInvalid':
+        return None
+
+    key = (id(enum_type), s)
+    if key not in ENUM_CACHE:
+        ENUM_CACHE[key] = enum_type[CAMEL_CASE_CAPITALS.sub(r'_\1',
+                                                            str(s)).upper()]
+    return ENUM_CACHE[key]
+
+
+CAPNP_ENUM_CACHE = {}
+
+
+def to_capnp_enum(enum_type, e):
+    key = (id(enum_type), e)
+
+    if key not in CAPNP_ENUM_CACHE:
+        # Convert from snake_case to camelCase.
+        parts = []
+        for idx, part in enumerate(e.name.split('_')):
+            if idx == 0:
+                parts.append(part.lower())
+            else:
+                parts.append(part.capitalize())
+        camel_case_e = "".join(parts)
+
+        CAPNP_ENUM_CACHE[key] = enum_type.__dict__[camel_case_e]
+
+    return CAPNP_ENUM_CACHE[key]
+
+
+def read_switch(sw):
+    timing = sw.timing
+    sizing = sw.sizing
+
+    return graph2.Switch(
```

Comment:
Why are the `graph2.Switch` object and the captnproto object are not compatible?

Could we make the attribute names for these objects in `graph2` just match the captnproto versions?

Reply:
The whole point of graph2 objects is to be storage independent.  As for field names, the capnp naming is camelCase, which non of the rest of our python code is.